ci/run-dockerized.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u -x
+
+image_name=$1
+
+if [[ ${2-} == "--dry-run" || ${CIRCLE_BRANCH-} != "master" ]]; then
+  do_dput=false
+else
+  do_dput=true
+fi
+
+DIRECTORY_HASH=$(git rev-parse HEAD:ci/$image_name)
+export DIRECTORY_HASH
+cd ci/$image_name/
+
+# If there is no cached image for the expected Git&Python versions and the current state of ci/tox,
+# build the image and push it to the Docker Hub.
+docker-compose --ansi never pull $image_name
+# A very unpleasant workaround for https://github.com/docker/compose/issues/7258
+# (since v1.25.1, `docker-compose pull` is NOT failing when it can't fetch the image).
+image_tag=$(docker-compose config | yq eval ".services.$image_name.image" -)
+docker image inspect "$image_tag" &>/dev/null || {
+  docker-compose build --build-arg user_id="$(id -u)" --build-arg group_id="$(id -g)" $image_name
```

Comment:
Nit: `--ansi never` (unless there're no weird artifacts in the CI output, but IIRC there are)

Reply:
Btw FYI, `ansi` here is a mental shortcut for https://en.wikipedia.org/wiki/ANSI_escape_code ... there are zillions of them, but the most important ones are for colors (we use a few of them in `constants.py`)